packages/react-sdk/src/App.tsx

```diff
@@ -25,7 +25,7 @@ export type AppProps = {
 };
 
 export const App = ({ layoutProps }: AppProps) => {
-  const { t } = useTranslation();
+  const { t } = useTranslation('neoboard');
```

Comment:
So the thing is that the default namespace is for the current i18n namespace. It works for the widget itself of course but for any application using the react-sdk they will likely have their own default namespace which is different. Meaning we sadly have to do this for it to be consumable. The default namespace is essentially shared across all dependencies consuming the i18n project. Its not scoped per project.

Reply:
I can see the default being set in the widget's init, am not sure what it's needed in the react-sdk exports for, but I get why it would need to be hardcoded somewhere in the react-sdk.

Reply:
> I can see the default being set in the widget's init, am not sure what it's needed in the react-sdk exports for, but I get why it would need to be hardcoded somewhere in the react-sdk.

You can either set it on the i18n init which sets it for the whole i18n instance. Thats whats done on the widget.

Then you can set it on each hook which is local for the hook. This is required as we cant assume the default namespace of consumers. The widget is one consumer but it isn't the only consumer. So we have to make sure that it is still using the neoboard namespace for neoboard translations regardless of the default namespace. We can (arguably) remove the default namespace on the widget itself. Its just there since it would kinda make sense. It wont have any significant effect however.

TLDR:

We cant assume things about how the i18n instance of consumers is set up. Hence we need to use i18next in predictable ways instead.

Reply:
![grafik](https://github.com/user-attachments/assets/6114f003-5feb-4ff7-acd7-4a6b84a8f8c2)

Graphic that hopefully makes this a bit clearer.

So both things always access one i18n instance. However the namespaces may be different